nix/pkgs/devshell.nix

```diff
@@ -0,0 +1,18 @@
+{
+  perSystem = {pkgs, ...}: {
+    devShells.default = with pkgs;
+      mkShell {
+        packages = lib.flatten [
+          deno
+          nodejs
+          cargo
+          protobuf
+          cmake
+          (lib.optionals stdenv.hostPlatform.isLinux [
+            libxkbcommon.dev
+            gtk-layer-shell
```

Comment:
Is this actually used somewhere?

Reply:
I made a mistake with `gtk-layer-shell` before when trying to support Wayland and left this artifact in. Really I think I just need `wayland wayland-protocols wlr-protocols` dependencies as in `overlay.nix`. And the devShell was provided for developing the project easier in case anyone wants to use Nix to standardize the tooling in development and not install stuff globally on their system.